pons/routing/router.py

```diff
@@ -98,10 +103,28 @@ def on_scan_received(self, msg: pons.Message, remote_id: int):
             # self.log("DUP PEER: %d" % remote_id)
 
     def on_peer_discovered(self, peer_id):
-        self.log("peer discovered: %d" % peer_id)
+        #self.log("peer discovered: %d" % peer_id)
+        self.netsim.event_manager.on_peer_discovery(self.my_id, peer_id)
 
     def on_msg_received(self, msg: pons.Message, remote_id: int):
-        self.log("msg received: %s from %d" % (msg, remote_id))
+        self.netsim.routing_stats['relayed'] += 1
+        if not self.is_msg_known(msg):
+            self.remember(remote_id, msg)
+            msg.hops += 1
+            self.store_add(msg)
+            if msg.dst == self.my_id:
+                # print("msg arrived", self.my_id)
+                self.netsim.routing_stats['delivered'] += 1
+                self.netsim.routing_stats['hops'] += msg.hops
+                self.netsim.routing_stats['latency'] += self.env.now - msg.created
+                self.netsim.event_manager.on_message_delivered(self.my_id, remote_id, msg)
+            else:
+                # print("msg not arrived yet", self.my_id)
+                self.netsim.event_manager.on_message_received(self.my_id, remote_id, msg)
+                self.forward(msg)
```

Comment:
the standard router should not have any biz logic, so forwarding on its own is a no go.
the question is how can this be refactored to automatically send out the events to the event manager but at the same time keep the logic in the actual routing impls.

Reply:
but then again router impl get so much shorter, not sure what the most elegant solution is.